src/hooks/useAdBlockDetection.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import Cookies from "js-cookie";
+import { AD_BLOCKER_DISMISSED } from "constants/cookies";
+import { useToastContext } from "context/toast";
+import { environmentalVariables } from "utils";
+import { get } from "utils/request";
+
+const { isProduction } = environmentalVariables;
+// This url is what newrelic downloads to support analytics it gets blocked by adblockers
+const AD_URL = "https://js-agent.newrelic.com/nr-spa-1158.min.js";
+
+/** useAdBlockDetection is a hook that detects if a user has an adblocker enabled. It checks this by trying to download
+ * a file from a url that is blocked by adblockers. If the file is not downloaded, the user has an adblocker enabled.
```

Comment:
This is an interesting approach. I like it, although I believe we run the risk of mistakenly labeling everyone as having Adblock enabled if this URL ever starts returning an error for whatever reason. However, I think that's a reasonable tradeoff -- it would be easy enough to address if it comes up. 🎉

Reply:
Yup this was one of the few files i could try to download that failed with adblock. That doesn't involve me actually downloading some tracking script. And/or trying to hit some tracking endpoint. It seemed like the most innocent approach.